### ISSUE TITLE
Is it possible to run tests without Stack?

### ISSUE THREAD

**Issue by Edwin-Pratt** [exercism/haskell#947]
Hey there. :innocent:

I've recently gotten started with Haskell and have thought about doing some Exercism challenges to practice my knowledge so far. However, I'm stuck by the "Hello, World" challenge because I'm not sure how to run tests without Stack... My Haskell setup is as follows:

- GHC - `8.10.3`
- Cabal - `3.2.0.0`
- OS - `Windows 10, 64-bit

When I install the "Full Haskell Platform", it didn't come with Stack.

There's a `cabal test` command, but it expects that there's a `cabal.project` file in the directory, and unfortunately, I only see `stack.yaml` and `package.yaml` when I've downloaded the project files...

**Comment by sshine** [comment]
Hi Edwin!

Currently we only ship a package.yaml with each exercise and not a .cabal file. This format is called [Hpack](https://github.com/sol/hpack). `stack` understands this, but `cabal` doesn't. So to answer your immediate question: No, right now you have to install Stack, too. You can install Stack on Windows using the 64-bit installer mentioned in [the Stack docs](https://docs.haskellstack.org/en/stable/README/).

We could ship both package.yaml and a .cabal file, or just a .cabal file, since Stack understands those, too. This was last brought up #892, which references when the decision was made and brings up pros and cons of the many choices available. If you have something to add to this discussion, that issue would be a good place.

If you're just looking for a solution, though, installing Stack is probably the shortest path. :)

Note that if you use Stack, it will actually download and manage copies of GHC that correspond to the `resolver` in stack.yaml for each of your projects. They're gonna live somewhere else than the GHC you currently installed (on Linux this is `~/.stack/`, so probably something equivalent to a sub-directory in your home directory on Windows), just so you know. If you want to use a GHC compatible with the project you're looking at, this is a quick way to accomplish that. Then all you have to do is run `stack ...`, e.g. `stack test` or `stack ghci`, to activate the right GHC version with the right set of external dependencies.

**Comment by sshine** [comment]
I want to quickly add that #892 contains:

> It is possible to generate a .cabal file with `cabal2nix` or `hpack`, [...]

You'd still need to have a GHC that is compatible with the one mentioned in stack.yaml.

For lts-16.21, which is the one all exercises currently ship with, this is GHC 8.8.

You probably won't have a problem with GHC 8.10. But you'd be among few who try. :)

If you choose to go this path, be sure to keep us posted on what does not work.

**Comment by Edwin-Pratt** [comment]
Hey there! :innocent:

Thanks so much for responding, I'll check Stack out! :smiley:

**Comment by petertseng** [comment]
I run tests without Stack. I consider it unsupported since it is not listed in the track's documentation. The command I use is: `runghc -Wall -isrc test/Tests.hs`. YMMV.

**Comment by Edwin-Pratt** [comment]
> I run tests without Stack. I consider it unsupported since it is not listed in the track's documentation. The command I use is: `runghc -Wall -isrc test/Tests.hs`. YMMV.

Hey there! 😇 

I've tried running `runghc -Wall isrc test/Tests.hs`, however I'm getting an error saying that GHC could not find `Test.HSpec`. I've followed the instructions on installing HSpec with `cabal update && cabal install hspec`, and even though it's installed globally, GHC can't find it.

---

I've managed to setup something in the meantime with `hpack`. If I run `hpack` in the directory containing `package.yaml`, it'll generate a `.cabal` file which can be used with `cabal test`. It's slow but `1 of 1 test suites (1 of 1 test cases) passed.` 😃 

Thanks for the help! 🚀
